src/btree/bt_random.c

```diff
@@ -365,14 +365,22 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags, WT_
 
     btree = S2BT(session);
     current = NULL;
-    retry = 100;
     /*
      * This function is called by eviction to find a random page in the cache. That case is
      * indicated by the WT_READ_CACHE flag. Ordinary lookups in a tree will read pages into cache as
      * needed.
      */
     eviction = LF_ISSET(WT_READ_CACHE);
 
+    if (eviction) {
+        retry = S2C(session)->evict_random_retries;
+        if (retry == 0) {
```

Comment:
retry equals 0 is weird. Does it make sense to just use a boolean to configure this?

Reply:
Yes that makes more sense, thanks @quchenhao. Updated in  [b584993](https://github.com/wiredtiger/wiredtiger/pull/10437/commits/b58499368b51168ef70bb868f03b6a1b9e252895)